qt-models/cylindermodel.cpp

```diff
@@ -237,6 +239,28 @@ QVariant CylindersModel::data(const QModelIndex &index, int role) const
 			return static_cast<int>(cyl->type.workingpressure.mbar);
 		case SIZE_INT:
 			return static_cast<int>(cyl->type.size.mliter);
+		case SENSORS: {
+			QStringList sensorStrings;
+			std::vector<int> allSensors(MAX_SENSORS);
+			std::iota(allSensors.begin(), allSensors.end(), 0);
+			for (int i = 0; i < current_dc->samples; ++i) {
+				auto &sample = current_dc->sample[i];
+				for (auto it = allSensors.begin(); it != allSensors.end();) {
+					int s = *it;
+					if (sample.pressure[s].mbar) {
```

Comment:
So now it'll search for the first sample with pressure information and see which cylinder the sensor is attached to.

I wonder if maybe the sensor array should be moved from sample to divecomputer. Does it ever make sense to have the same sensor change cylinder during a dive? Technically someone could remove the regulator from one cylinder and attach it to another during the dive, but it would be very rare and the cases with a sensor attached to the regulator likely even rarer. The file format could still stay as it is, even though it would let you store information that isn't used.

Reply:
So looking a bit more into this it looks like I got the data model for the sensors wrong. I treated the index in the sensor array as a sensor id containing the cylinder this sensor was connected to. This works for my use case of having two dive computers with one transmitter each, since the transmitter is always in the first index, but with more transmitter per dive computer this doesn't work. The sensor index has no real meaning, it just shows that the pressure at index X is for cylinder at sensor index X. So for one sample index 0 could contain cylinder 0 and index 1 cylinder 1, for the next sample index 0 could be cylinder 1 and index 1 cylinder 2.

Reply:
So, it actually seems to work in most cases with up to two sensors for a dive computer, since add_sample_pressure() tries to maintain the index.

Two exceptions seems to be the liquidvision and uemis parsers, that always puts the active sensor in index 0.